node_follower/src/txn_service.rs

```diff
@@ -26,13 +35,16 @@ impl TransactionService {
             txn: Some(txn),
             key: key.to_vec(),
         };
-        let res = self.client.submit(req).await?.into_inner();
+
+        let mut client = self.random_client()?;
+        let res = client.submit(req).await?.into_inner();
         Ok(res)
     }
 
     pub async fn query(&mut self, key: &[u8]) -> Result<TxnQueryRespV1> {
         let req = TxnQueryReqV1 { key: key.to_vec() };
-        let res = self.client.query(req).await?.into_inner();
+        let mut client = self.random_client()?;
```

Comment:
since you're querying the txn manager you submitted the txn to, i don't think randomly selecting a client to query is likely to get you back the same client that has the txn you're looking for in its queue